src/lib/IDBDAO.js

```diff
@@ -0,0 +1,377 @@
+/**
+ * @license
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+foam.CLASS({
+  package: 'foam.dao',
+  name: 'IDBException',
+  extends: 'foam.dao.ExternalException',
```

Comment:
Let's make this an InternalException and rename to IDBInternalException.  Most exceptions that indexeddb throws are likely internal.

Internal means there's something going wrong inside IDB, like it can't talk to the disk/network.  External means there's something wrong with the data you are trying to store, such as it doesn't pass validation, or you don't have permissions.